docs/guides/isolated_systems.md

```diff
@@ -0,0 +1,221 @@
+# Isolated systems guide
+
+SQLMesh assumes that production and non-production data systems can access one another, but that is not always the case.
```

Comment:
what about something like sqlmesh is optimized for use cases where developers have access to production data